nebula_hw_interfaces/include/nebula_hw_interfaces/nebula_hw_interfaces_hesai/hesai_cmd_response.hpp

```diff
@@ -251,157 +253,189 @@ struct HesaiInventory
+  }
+  return os;
+}
+
+struct HesaiConfig_XT_40p: public HesaiConfigBase
 {
-  big_uint32_buf_t system_uptime;
-  big_uint16_buf_t motor_speed;
-  big_int32_buf_t temperature[8];
-  uint8_t gps_pps_lock;
-  uint8_t gps_gprmc_status;
-  big_uint32_buf_t startup_times;
-  big_uint32_buf_t total_operation_time;
-  uint8_t ptp_clock_status;
-  uint8_t reserved[5];  // FIXME: 4 bytes labeled as humidity in OT128 datasheet
-
-  friend std::ostream & operator<<(std::ostream & os, nebula::HesaiLidarStatus const & arg)
+  struct Internal : HesaiConfigBase::Internal{
+    uint8_t clock_data_fmt;
+    uint8_t noise_filtering;
+    uint8_t reflectivity_mapping;
+    unsigned char reserved[6];
+  };
+
+  HesaiConfig_XT_40p(Internal value) : value(value) {}
+  [[nodiscard]] const HesaiConfigBase::Internal & get() const override
   {
-    std::ios initial_format(nullptr);
-    initial_format.copyfmt(os);
+    return value;
+  }
+  json sensor_specifics_to_json() override
+  {
+    json j;
+    j["clock_data_fmt"] = value.clock_data_fmt;
+    j["noise_filtering"] = value.noise_filtering;
+    j["reflectivity_mapping"] = value.reflectivity_mapping;
 
-    os << "system_uptime: " << arg.system_uptime;
-    os << ", ";
-    os << "motor_speed: " << arg.motor_speed;
-    os << ", ";
-    os << "temperature: ";
+    return j;
+  }
+  Internal value;
+};
 
-    for (size_t i = 0; i < sizeof(arg.temperature); i++) {
-      if (i != 0) {
-        os << ',';
-      }
-      os << arg.temperature[i];
-    }
+inline std::ostream & operator<<(std::ostream & os, HesaiConfig_XT_40p & arg)
+{
+  for (auto & [key, value] : arg.to_json().items()) {
+    os << key << " : " << value << std::endl;
+  }
+  return os;
+}
 
-    os << ", ";
-    os << "gps_pps_lock: " << static_cast<int>(arg.gps_pps_lock);
-    os << ", ";
-    os << "gps_gprmc_status: " << static_cast<int>(arg.gps_gprmc_status);
-    os << ", ";
-    os << "startup_times: " << arg.startup_times;
-    os << ", ";
-    os << "total_operation_time: " << arg.total_operation_time;
-    os << ", ";
-    os << "ptp_clock_status: " << static_cast<int>(arg.ptp_clock_status);
-    os << ", ";
-    os << "reserved: ";
+/// @brief struct of PTC_COMMAND_GET_LIDAR_STATUS
+struct HesaiLidarStatusBase
+{
+  struct Internal{
+    big_uint32_buf_t system_uptime;
+    big_uint16_buf_t motor_speed;
+  };
 
-    for (size_t i = 0; i < sizeof(arg.reserved); i++) {
-      if (i != 0) {
-        os << ' ';
-      }
-      os << std::hex << std::setfill('0') << std::setw(2) << (static_cast<int>(arg.reserved[i]));
-    }
-    os.copyfmt(initial_format);
+  json to_json()
+  {
+    json j;
+    j["system_uptime"] = std::to_string(get().system_uptime.value()) + " s";
+    j["motor_speed"] = std::to_string(get().motor_speed.value()) + " RPM";
+    j.update(sensor_specifics_to_json());
 
-    return os;
+    return j;
   }
+  
+  // protected:
+  virtual json sensor_specifics_to_json() = 0;
```

Comment:
For all structs, please make `sensor_specifics_to_json` `protected`. `protected` means that only the base and child classes can see and use the function. Since the function is an internal implementation detail, we don't want to expose it to the public.